pkg/auth/service.go

```diff
@@ -959,3 +964,626 @@ func (s *DBAuthService) Authorize(ctx context.Context, req *AuthorizationRequest
 	// we're allowed!
 	return &AuthorizationResponse{Allowed: true}, nil
 }
+
+type APIAuthService struct {
+	apiClient   *ClientWithResponses
+	secretStore crypt.SecretStore
+	cache       Cache
+}
+
+func (a *APIAuthService) SecretStore() crypt.SecretStore {
```

Comment:
I think the only thing the service needs to expose is the shared secret. Can be handled in a different PR.